mlir/lib/Dialect/Bufferization/Transforms/BufferDeallocationSimplification.cpp

```diff
@@ -49,19 +49,53 @@ static LogicalResult updateDeallocIfChanged(DeallocOp deallocOp,
   return success();
 }
 
-/// Checks if 'memref' may or must alias a MemRef in 'memrefList'. It is often a
+/// Given a memref value, return the "base" value by skipping over all
+/// ViewLikeOpInterface ops (if any) in the reverse use-def chain.
+static Value getViewBase(Value value) {
+  while (value) {
+    auto viewLikeOp = value.getDefiningOp<ViewLikeOpInterface>();
+    if (!viewLikeOp)
+      return value;
+    value = viewLikeOp.getViewSource();
+  }
+  return value;
```

Comment:
`value` shouldn't ever be a null value, right? Couldn't we do the following then:
```suggestion
  while (auto viewLikeOp = value.getDefiningOp<ViewLikeOpInterface>())
    value = viewLikeOp.getViewSource();
  return value;
```